platform/android/display_server_android.cpp

```diff
@@ -569,7 +569,9 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
 #endif
 
 	Input::get_singleton()->set_event_dispatch_function(_dispatch_input_events);
+
 	Input::get_singleton()->set_use_input_buffering(true); // Needed because events will come directly from the UI thread
+	Input::get_singleton()->set_has_input_thread(true);
```

Comment:
Knowing why these are not exclusive would help me understand the new approach better.

Reply:
Maybe I can add a note about this.

It is rather confusing but is historical - accumulated input was added as an afterthought, and it overrides the `use_input_buffering` setting. I did attempt to change this to something more sensible, but it would have been compat breaking. (This is why the unit tests were failing when I first made the PR, so I reverted to logic mirroring the legacy.)

If you look in the `_update_buffering_mode()` logic you can see that buffering will still be used if `use_input_buffering` is false (!) if `use_accumulated_input` is true. This is because accumulated input requires buffering to work. This is how the legacy logic works.

If we are prepared to break compatibility, then we can probably make some simpler exclusive logic.

Reply:
Have added some comments in the `_update_buffering_mode()` function to this effect.